addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairEasyTouchHandler.java

```diff
@@ -0,0 +1,426 @@
+                case EASYTOUCH_AUX4:
+                    circuitSwitch(5, state);
+                    break;
+                case EASYTOUCH_AUX5:
+                    circuitSwitch(7, state);
+                    break;
+                case EASYTOUCH_AUX6:
+                    circuitSwitch(8, state);
+                    break;
+                case EASYTOUCH_AUX7: // A5 01 10 20 86 02 09 01
+                    circuitSwitch(9, state);
+                    break;
+            }
+        } else if (command instanceof DecimalType) {
+            int sp = ((DecimalType) command).intValue();
+
+            switch (channelUID.getId()) {
+                case EASYTOUCH_SPASETPOINT:
+                    setPoint(false, sp);
+                    break;
+                case EASYTOUCH_POOLSETPOINT:
+                    setPoint(true, sp);
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Method to turn on/off a circuit in response to a command from the framework
+     *
+     * @param circuit - circuit number
+     * @param state
+     */
+    public void circuitSwitch(int circuit, boolean state) {
+        byte[] packet = { (byte) 0xA5, (byte) 0x01, (byte) m_id, (byte) 0x00 /* source */, (byte) 0x86, (byte) 0x02,
+                (byte) circuit, (byte) ((state) ? 1 : 0) };
+
+        PentairPacket p = new PentairPacket(packet);
+
+        PentairBaseBridgeHandler bbh = (PentairBaseBridgeHandler) this.getBridge().getHandler();
+        bbh.writePacket(p);
+    }
+
+    /**
+     * Method to set heat point for pool (true) of spa (false)
+     *
+     * @param Pool pool=true, spa=false
+     * @param temp
+     */
+    public void setPoint(boolean Pool, int temp) {
+        // [16,34,136,4,POOL HEAT Temp,SPA HEAT Temp,Heat Mode,0,2,56]
+        // [165, preambleByte, 16, 34, 136, 4, currentHeat.poolSetPoint, parseInt(req.params.temp), updateHeatMode, 0]
+        int spaset = (!Pool) ? temp : phspcur.spasetpoint;
+        int poolset = (Pool) ? temp : phspcur.poolsetpoint;
+        int heatmode = (phspcur.spaheatmode << 2) | phspcur.poolheatmode;
+
+        byte[] packet = { (byte) 0xA5, (byte) 0x01, (byte) m_id, (byte) 0x00 /* source */, (byte) 0x88, (byte) 0x04,
+                (byte) poolset, (byte) spaset, (byte) heatmode, (byte) 0 };
+
+        logger.info("Set {} temperature: {}", (Pool) ? "Pool" : "Spa", temp);
+
+        PentairPacket p = new PentairPacket(packet);
+
+        PentairBaseBridgeHandler bbh = (PentairBaseBridgeHandler) this.getBridge().getHandler();
+        bbh.writePacket(p);
+    }
+
+    @Override
+    public void processPacketFrom(PentairPacket p) {
+        switch (p.getAction()) {
+            case 1: // Write command to pump
+                logger.trace("Write command to pump (unimplemented): {}", p.toString());
+                break;
+            case 2:
+                if (p.getLength() != 29) {
+                    logger.debug("Expected length of 29: ", p.toString());
```

Comment:
You don't have to call toString it will be called automatically, but only when the logging-level actually applies.